src/globus_sdk/services/auth/_common.py

```diff
@@ -0,0 +1,31 @@
+from __future__ import annotations
+
+from globus_sdk._types import ScopeCollectionType
+from globus_sdk.exc import GlobusSDKUsageError
+from globus_sdk.exc.warnings import warn_deprecated
+from globus_sdk.scopes import AuthScopes, MutableScope, TransferScopes
+
+_DEFAULT_REQUESTED_SCOPES = (
+    AuthScopes.openid,
+    AuthScopes.profile,
+    AuthScopes.email,
+    TransferScopes.all,
+)
+
+
+def stringify_requested_scopes(requested_scopes: ScopeCollectionType | None) -> str:
+    if requested_scopes is None:
+        warn_deprecated(
+            "`requested_scopes` was not specified or was given as `None`. "
+            "A default set of scopes will be used, but this behavior is deprecated. "
+            "Specify an explicit set of scopes instead.",
+            stacklevel=3,
+        )
+        requested_scopes = _DEFAULT_REQUESTED_SCOPES
```

Comment:
We haven't talked about this, so there may be mixed feelings about deprecating this behavior, but I think it's for the best in the long run.

The current situation is that the interfaces for users look akin to
```python
tokens = do_login()
... = do_stuff_to_tokens(tokens)
tc = TransferClient(...)  # works
gc = GroupsClient(...)  # fails
```

and it's not obvious why this is the case.

Of course, we could keep growing and changing the default scopes, but that also has serious problems.

I think we ought to remove this in v4 and this puts us on a path in which we can start updating docs and educating users on what we want them to do (which is, I think, to use `globus_sdk.scopes`).

Reply:
Agree with this change.

I do think it'll be useful to keep in mind (read out-of-scope for this PR) as we go forward how customers should be referencing common groups of scopes. Are there enough common groups of scopes that aren't static scopes to make that into a useful SDK feature?

Mostly just concerned about SDK customers needing to learn too much down the scope trail before they can start using a simple api invocation.